### PR TITLE
Retry at errors at image pull, properly fail if retries fail

### PR DESCRIPTION
**Pull request by vienin**

### CODE REVIEW

File: pkr/driver/base.py [hunk 1]
```diff
@@ -190,32 +195,53 @@ def pull_images(self, services, registry, tag=None):
...
+                             tag=tag)
+
+            # Strip the repository tag
+            self.docker.tag(image=':'.join((rep_tag, tag)),
+                            repository=image_name,
+                            tag=tag,
+                            force=True)
+
+        except docker.errors.APIError as error:
+            error_smg = 'Error while pulling the image {}: {}'.format(
```
Comment (popawu) [inline pkr/driver/base.py]: error_msg ?

Comment (vienin) [reply pkr/driver/base.py]: Fixed !

File: requirements.txt [hunk 2]
```diff
@@ -27,6 +27,7 @@ scandir==1.9.0
 six==1.11.0
 smmap2==2.0.5
 stevedore==1.21.0
+tenacity==5.0.3
```
Comment (Kimamisa) [inline requirements.txt]: This needs to be added to the setup.py file too.

Comment (vienin) [reply requirements.txt]: Fixed !

**Pull request merged**